File: src/main/java/io/github/shaksternano/borgar/media/io/writer/MediaWriter.java
package io.github.shaksternano.borgar.media.io.writer;

import io.github.shaksternano.borgar.media.AudioFrame;
import io.github.shaksternano.borgar.media.ImageFrame;

import java.io.Closeable;
import java.io.IOException;

public interface MediaWriter extends Closeable {

    void writeImageFrame(ImageFrame frame) throws IOException;

    void writeAudioFrame(AudioFrame frame) throws IOException;

    boolean isStatic();

    boolean supportsAudio();
}

File: core/src/main/kotlin/io/github/shaksternano/borgar/core/media/writer/MediaWriter.kt
package io.github.shaksternano.borgar.core.media.writer

import io.github.shaksternano.borgar.core.media.AudioFrame
import io.github.shaksternano.borgar.core.media.ImageFrame
import java.io.Closeable

interface MediaWriter : Closeable {

    val isStatic: Boolean
    val supportsAudio: Boolean

    suspend fun writeImageFrame(frame: ImageFrame)

    suspend fun writeAudioFrame(frame: AudioFrame)
}
